### DISCUSSION TITLE
OpenAPI description of the REST API

### DISCUSSION THREAD

**Discussion by klaernie** [sni/thruk#1415] [Ideas]
Hi there!

I was wondering if anybody so far has thought about rewriting the REST API documentation as an OpenAPI.

For work I'm currently writing some stub OpenAPI for our Nagios installation, and I was wondering if there would be interest in transferring the current documentation into an OpenAPI description. For now I'm just pointing to https://thruk.org/documentation/rest.html internally, but I was contemplating converting the docs so people can use interactive try-out utilities to toy with the API and generate code to access the API automatically.

Of cause I cannot guarantee that the result is the best possible with OpenAPI, although I'd try my hardest ;)

Are there any particular opinions on OpenAPI docs?

Best regards,
Andre

**Comment by sni** [comment]
I haven't used openapi docs yet, but the current rest api documentation is auto-generated. So depending on how complicated openapi docs are, it might simply be generated along the other docs.
